src/app/selectors/manuscript-editor.selectors.ts

```diff
@@ -0,0 +1,44 @@
+import { createSelector } from 'reselect';
+import { ApplicationState, ManuscriptEditorState } from '../store';
+import { get } from 'lodash';
+import { EditorState } from 'prosemirror-state';
+import { getManuscriptData } from './manuscript.selectors';
+
+const getManuscriptEditorState = (state: ApplicationState): ManuscriptEditorState => {
+  return state.manuscriptEditor;
+};
+
+export const getFocusedEditor = createSelector(
+  getManuscriptEditorState,
+  getManuscriptData,
+  (manuscriptEditorState, manuscriptState) => {
+    const path = get(manuscriptEditorState, 'focusedManuscriptPath');
+    return get(manuscriptState, ['present', path].join('.'));
+  }
+);
+
+export const canItalicizeSelection = createSelector(getFocusedEditor, (editorState: EditorState) => {
+  if (editorState) {
+    return editorState.schema.marks.italic && !editorState.selection.empty;
+  }
+  return false;
+});
+
+export const canUndoChanges = createSelector(getManuscriptData, (data) => get(data, 'past.length') > 0);
+
+export const canRedoChanges = createSelector(getManuscriptData, (data) => get(data, 'future.length') > 0);
+
+export const canBoldSelection = createSelector(getFocusedEditor, (editorState: EditorState) => {
```

Comment:
Whilst testing, the bold button isn't activating for me. Should it? If so then might be something wrong in the toolbar code.

Reply:
So what selectors do is check the schema of a particular editorState. If bold is part of the schema then it will highlight. If it's not then it won't. See `editorState.schema.marks.bold`